### ISSUE TITLE
2.0.0-rc2 - simple example - MSVC 12 hangs while 14/15 dont

### ISSUE THREAD

**Issue by solvingj** [log4cplus/log4cplus#296]
We've got a very simple example that we use as a package test.  It worked fine under 1.2.1 (all compilers), and it works fine on `2.0.0-rc2` under Clang, GCC, Apple-Clang, and MSVC14 and 15.  However, under MSVC12, it hangs.  It builds and runs, and does not raise an exception, but it hangs.  Please let us know if you are even interested in supporting MSVC12, and if not, we can remove support for it from the package and close the ticket. Otherwise, info is below. 

Sample code. 
```cpp
#include <log4cplus/helpers/loglog.h>
#include <log4cplus/logger.h>

using namespace log4cplus::helpers;

int main()
{
    LogLog::getLogLog()->debug(LOG4CPLUS_TEXT("This is a Debug statement..."));
    LogLog::getLogLog()->debug(
        log4cplus::tstring(LOG4CPLUS_TEXT("This is a Debug statement...")));

    return 0;
}
```

https://ci.appveyor.com/project/BinCrafters/conan-log4cplus

**Comment by wilx** [comment]
The 2.x series is different. You need the initializer: See https://github.com/log4cplus/log4cplus#windows-and-visual-studio

**Comment by solvingj** [comment]
Ok, right, so i tried that and get this error:

```
         test_package.cpp
         You are not using DLL C run time library. You must call log4cplus::initialize() once before you use any other log4cplus API.
    
``` 
https://ci.appveyor.com/project/BinCrafters/conan-log4cplus/build/1.0.38/job/6c9t4swlfad2tdvh#L1539

**Comment by wilx** [comment]
> You are not using DLL C run time library. You must call log4cplus::initialize() once before you use any other log4cplus API.

This should probably removed. It is true for older series but 2.0.x relies on `log4cplus::Initializer`.

You actually have a compile error there. You have to `#include <log4cplus/initializer.h>` and declare `log4cplus::Initializer initializer;` as the very first thing after main's opening brace. See, e.g., https://github.com/log4cplus/log4cplus/blob/master/tests/unit_tests/unit_tests.cxx#L6.

**Comment by solvingj** [comment]
Fixed, thanks!  Any estimation on 2.0.0 release?
